go-controller/pkg/ovn/egressfirewall_dns.go

```diff
@@ -167,14 +167,16 @@ func (e *EgressDNS) Run(defaultInterval time.Duration) {
 	// initially the next DNS Query happens at the default interval
 	durationTillNextQuery := defaultInterval
 	go func() {
+		timer := time.NewTicker(durationTillNextQuery)
+		defer timer.Stop()
 		for {
 			// perform periodic updates on dnsNames as each ttl runs out, checking for updates at
 			// least every defaultInterval. Update durationTillNextQuery everytime a new DNS name gets
 			// added
 			select {
 			case <-e.added:
 				//on update need to check if the GetNextQueryTime has changed
-			case <-time.After(durationTillNextQuery):
+			case <-timer.C:
 				if len(dnsName) > 0 {
```

Comment:
How about we replace `dnsName` with `domainNameExpiringFirst` or anything else that captures the idea that this is the one domain name in our list that will expire before all others.